mmv1/products/dlp/api.yaml

```diff
@@ -1602,3 +1602,713 @@ objects:
                                       - :ALPHA_LOWER_CASE
                                       - :PUNCTUATION
                                       - :WHITESPACE
+                        - !ruby/object:Api::Type::NestedObject
+                          name: 'cryptoReplaceFfxFpeConfig'
+                          description: |
+                            Replaces an identifier with a surrogate using Format Preserving Encryption (FPE) with the FFX mode of operation; however when used in the `content.reidentify` API method, it serves the opposite function by reversing the surrogate back into the original identifier. The identifier must be encoded as ASCII. For a given crypto key and context, the same identifier will be replaced with the same surrogate. Identifiers must be at least two characters long. In the case that the identifier is the empty string, it will be skipped. See [https://cloud.google.com/dlp/docs/pseudonymization](https://cloud.google.com/dlp/docs/pseudonymization) to learn more.
+
+                            Note: We recommend using CryptoDeterministicConfig for all use cases which do not require preserving the input alphabet space and size, plus warrant referential integrity.
+                          properties:
+                            - !ruby/object:Api::Type::NestedObject
+                              name: 'cryptoKey'
+                              description: |
+                                The key used by the encryption algorithm.
+                              properties:
+                                - !ruby/object:Api::Type::NestedObject
+                                  name: 'transient'
+                                  description: |
+                                    Transient crypto key
```

Comment:
Agreed - I'll update the file with the richer descriptions.

Something I was thinking when working on this PR was how repetitive it was, with entities like [CryptoKey](https://cloud.google.com/dlp/docs/reference/rest/v2/projects.deidentifyTemplates#cryptokey) appearing 6 times, and [Value](https://cloud.google.com/dlp/docs/reference/rest/v2/Value) appearing 8 times in the file. Is there any value in storing the yaml for those entities in a separate file and then templating it into api.yaml?

Reply:
Hypothetically yes; however, apparently that can lead to subtle issues because sometimes even if the docs are identical the actual behavior of the API fields can be slightly different - for example, they might be required in some cases and optional in others. This is apparently something that's been tried in the past and deemed not worth it. (Not to say that decision couldn't be revisited, just to say it's where we're at today.)

Reply:
I figured it was something that would have been considered previously, makes a lot of sense. Thanks for the context!